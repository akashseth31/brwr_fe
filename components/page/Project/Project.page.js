export default function ProjectPage({ name = '', version = 0, upgradeVesionFn = null }) {
  return (
    <>
      <h1>{name} version {version}</h1>
      <button onClick={upgradeVesionFn}>Upgrade Version</button>
    </>
  );
}
