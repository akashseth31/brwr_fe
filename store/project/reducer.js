import { UPGRADE_VERSION } from './actionTypes';

export const initialState = {
  name: 'Borrower platform',
  version: 1,
}

const reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case UPGRADE_VERSION:
      return { ...state, version: payload };
    default:
      return state;
  }
};

export default reducer;
