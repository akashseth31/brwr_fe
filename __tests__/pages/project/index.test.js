import { screen } from '@testing-library/react';
import Project from '../../../pages/project/index';
import '@testing-library/jest-dom';
import reducer, { initialState as project } from '../../../store/project/reducer';
import { renderWithState } from '../../../utils/render-wrapper.util';

const initialState = {
  project,
};

describe('Project', () => {
  let component;

  beforeEach(() => {
    component = renderWithState(<Project />, reducer, { initialState });
  });

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders a heading', () => {
    const heading = screen.getByRole('heading', {
      name: `${project.name} version ${project.version}`,
    })

    expect(heading).toBeInTheDocument();
  });

  it('renders a button', () => {
    const button = screen.getByRole('button', {
      name: 'Upgrade Version',
    });

    expect(button).toBeInTheDocument();
  });
});
