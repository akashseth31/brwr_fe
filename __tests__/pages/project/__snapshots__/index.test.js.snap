// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    Borrower platform version 1
  </h1>
  <button>
    Upgrade Version
  </button>
</DocumentFragment>
`;
